extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -55,12 +69,50 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		this._onReadAllTOCFiles.fire();
 	}
 
+	async openBook(bookPath: string, context: vscode.ExtensionContext, openAsUntitled: boolean): Promise<void> {
```

Comment:
Why is this async? It doesn't seem to be awaiting on anything. 

I'd suggest leaving it as async but then awaiting the call to getTableOfContentFiles and removing the .then.